lib/searchjoy/conversions.rb

```diff
@@ -0,0 +1,64 @@
+module Searchjoy
+  # smart reindexing of conversions
+  module Conversions
+    def self.reindex(options = {})
+      return {} if conversions_objects.empty?
+
+      batch_size = nil
+      callback = :bulk
+
+      options.keys.each do |key|
+        case options[key]
+        when :debug
+          log_output = Logger.new(STDOUT)
+
+          case options[:debug]
+          when :active_record
+            ActiveRecord::Base.logger = log_output
+          when :searchkick
+            Searchkick::LogSubscriber.logger = log_output if defined?(Searchkick)
+          when true
+            ActiveRecord::Base.logger = log_output
+            Searchkick::LogSubscriber.logger = log_output if defined?(Searchkick)
+          end
+        when :batch_size
+          batch_size = options[:batch_size]
+        when :callback
+          callback = options[:callback]
+        end
+      end
+
+      reindex_map.keys.each do |klass_name|
+        klass = klass_name.constantize
+
+        batch_size ||= klass.searchkick_options[:batch_size] if klass.respond_to?(:searchkick_options)
+        batch_size ||= 1_000
+
+        Searchkick.callbacks(callback) do
```

Comment:
This should go inside `find_in_batches` - otherwise, it'll queue up all the reindexes and do them once at the end.